### PR TITLE
Add defaults tests for restJson1

### PR DESCRIPTION
**Pull request by mullermp**
#### Background
Add defaults tests for restJson1 similar to AwsJson1_0 and CBOR

#### Testing
In smithy-ruby, not sure how to test here.

#### Links
n/a

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by syall — CHANGES_REQUESTED** [review]

File: smithy-aws-protocol-tests/model/restJson1/defaults.smithy [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        tags: ["defaults"]
+        protocol: restJson1
+        method: "POST"
+        bodyMediaType: "application/json"
+        uri: "/OperationWithDefaults"
+        headers: {"Content-Type": "application/json"}
+        body: """
+            {
+                "defaults": {
+                    "default_string": "hi",
```
[the remark this answers is not in the document]
Comment (mullermp) [reply smithy-aws-protocol-tests/model/restJson1/defaults.smithy:23]: Yes, they should. I will change it.

File: smithy-aws-protocol-tests/model/restJson1/defaults.smithy [hunk 2]
```diff
@@ -0,0 +1,371 @@
+$version: "2.0"
+
+namespace aws.protocoltests.restjson
+
+use aws.protocols#restJson1
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+apply OperationWithDefaults @httpRequestTests([
```
[the remark this answers is not in the document]
Comment (mullermp) [reply smithy-aws-protocol-tests/model/restJson1/defaults.smithy:9]: It didn't apply to us. I can add them too.

[the remark this answers is not in the document]
Comment (alextwoods) [reply smithy-aws-protocol-tests/model/restJson1/defaults.smithy:9]: I've added the server tests.

File: smithy-aws-protocol-tests/model/restJson1/defaults.smithy (hunk 1 above)
[the remark this answers is not in the document]
Comment (alextwoods) [reply smithy-aws-protocol-tests/model/restJson1/defaults.smithy]: I've updated all of the serialized member names.

**Comment by mullermp** [comment]
@syall Alex made the changes while I am away. Please re-review.

**Review by milesziemer — APPROVED** [review]

**Comment by mullermp** [comment]
Thank you!

**Review by syall — APPROVED** [review]

**Pull request merged**